en/docs/devops-and-ci-cd/configure-vpns-on-the-choreo-cloud-data-plane.md

```diff
@@ -0,0 +1,210 @@
+4. Click **Next**.
+5. In the **File Mount** pane that opens, click **+ Add**.
+6. To mount a configuration file to the Tailscale proxy component and specify the port mapping for the TCP forward proxy running there, do the following:
+    1. Specify `/Config.yaml` as the **Mount Path**.
+    2. Specify the following in the sample configuration file:
+       ```
+       portMappings:
+           8080: "100.108.78.93:8090"
+           8081: "100.108.78.93:1433"
+       ```
+7. Click **Next**.
+8. In the **Endpoints** pane that opens, click **+ Add** and edit the `endpoints.yaml` configuration to expose your Tailscale proxy as a service. The following is a sample `endpoints.yaml` configuration you can use:
+
+    !!! note
+        The sample `endpoints.yaml` file given below defines two project-level endpoints. These endpoints can be used by other components within the same project to access the services. If you want to directly expose your private endpoint via the Choreo gateway either with the **Public** or **Organization** visibility, you can set the `networkVisibility` property of the endpoint to `Public` or `Organization`.
+
+    ``` yaml
+
+    version: 0.1
+    endpoints:
+      - name: Private HTTP service
+        port: 8080
+        type: REST
+        networkVisibility: Project
+        context: /
+      - name: Private DB service
+        port: 8081
+        type: TCP
+        networkVisibility: Project
+        context: /
+
+    ```
+
+9. Click **Save**.
+10. Click **Next** and then click **Deploy**.
+
+    !!! note
+        Deploying the component may take a while. You can track the progress by observing the logs. Once the deployment is complete, the build status changes to **Active** on the **Development** environment card.
+
+When the component is deployed, you can observe a new node connected to your Tailscale network. To view this, go to the [Tailscale machines](https://login.tailscale.com/admin/machines) page of your Tailscale coordination server.
+
+### Step 3: Access private network endpoints with the Tailscale proxy
+
+Now you have successfully deployed the Tailscale proxy in your project and it is connected to your Tailnet. You can proceed to use the Tailscale proxy to provision access for other components to securely access private network endpoints.
+
+You can [configure endpoints](https://wso2.com/choreo/docs/develop-components/configure-endpoints/#learn-the-endpointsyaml-file) of the Tailscale proxy to use it for various aspects within Choreo.
+
+## Post-deployment actions
+
+### Handle node key expiry
+
+Tailscale nodes have a default [node key](https://tailscale.com/kb/1010/node-keys) expiry time of 180 days. Nodes require re-authentication after key expiry to avoid connection losses and application downtime. Consider disabling node key expiry if needed. For more details,  
```

Comment:
```suggestion
Tailscale nodes have a default [node key](https://tailscale.com/kb/1010/node-keys) expiry time of 180 days. Nodes require re-authentication after key expiry to avoid connection losses and application downtime. Consider disabling node key expiry if needed.  
```